### PR TITLE
Add commands to handle the trusted servers from command line

### PR DESCRIPTION
**Pull request by jvillafanez**
## Description
New commands to handle trusted servers from the command line

## Related Issue
https://github.com/owncloud/enterprise/issues/5731

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareCreateSpecialToShares2-maria10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/38474/52

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareCreateSpecialToShares1-maria10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/38474/50

File: apps/federation/lib/Command/TrustedServerRemove.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	 */
+	public function __construct(TrustedServers $trustedServers) {
+		parent::__construct();
+		$this->trustedServers = $trustedServers;
+	}
+
+	protected function configure() {
+		$this
+			->setName('federation:trusted-servers:remove')
+			->setDescription('List the trusted servers')
```
Comment (phil-davis) [inline apps/federation/lib/Command/TrustedServerRemove.php:42]:
```suggestion
			->setDescription('Remove a trusted server')
```

File: apps/federation/lib/Command/TrustedServerRemove.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$id = (int)$input->getArgument('id');
+		$this->trustedServers->removeServer($id);
```
Comment (phil-davis) [inline apps/federation/lib/Command/TrustedServerRemove.php:57]: What happens if the user passes an id that does not exist?
`removeServer` should complain somehow, and the command should then say something to the user and exit with non-zero status.

Comment (jvillafanez) [reply apps/federation/lib/Command/TrustedServerRemove.php:57]: The `removeServer` throws an exception:
```
> occ federation:trusted-servers:remove 88

In DbHandler.php line 121:
                               
  No Server found with ID: 88  
                               

federation:trusted-servers:remove <id>


> echo $?
1
```
Anyway, better to provide a formatting more aligned with the rest of the commands.

**Review by phil-davis — APPROVED** [review]
LGTM - I suppose that others should also review this.

File: apps/federation/lib/Command/TrustedServerRemove.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$id = (int)$input->getArgument('id');
+		try {
+			$this->trustedServers->removeServer($id);
```
Comment (pako81) [inline apps/federation/lib/Command/TrustedServerRemove.php:58]: Maybe add here ´$output->writeln("Removed server with id {$id}");´ to be consistent with the add command and to give some feedback to the user.

**Review by pako81 — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Comment by mrow4a** [comment]
@jvillafanez have a look also in https://github.com/owncloud/core/pull/40815 and https://github.com/owncloud/enterprise/issues/5676#issuecomment-1563949556

**Review by mrow4a — APPROVED** [review]

**Comment by pako81** [comment]
merge this?

**Pull request merged**